psi4/src/psi4/libfock/jk.cc

```diff
@@ -103,7 +103,9 @@ std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_
 
     } else if (jk_type == "MEM_DF") {
         MemDFJK* jk = new MemDFJK(primary, auxiliary);
+        jk->set_wcombine(true);
         _set_dfjk_options<MemDFJK>(jk, options);
+        if (options["WCOMBINE"].has_changed()) { jk->set_wcombine(options.get_bool("WCOMBINE")); }
```

Comment:
I think this should be plain `jk->set_wcombine(options.get_bool("WCOMBINE")` w/o the `if` clause. if you switch that here, do you still need the options line in the driver?

Reply:
I'm sure your suggestion will work, but I'm hesitant to implement it because all the other options in `jk.cc` are handled as line 108 above.

To be completely honest, I don't know whether or not I need the options line in the driver to begin with. I put it there to match the convention with handling of `do_wK`

Reply:
the other options in jk.cc are that way b/c the true default is set https://github.com/psi4/psi4/pull/1911/files#diff-822f8557d151459bb012892a642889b8R189-R207, not from read_options.

do_wK in the scf_iterations is querying `functional`, not globals. I'm not seeing a case for this line in the driver.

Reply:
If you don't see the case, the change can be made. My aesthetic preferences may be what they are, but I can go with what you think is less fragile or easier to follow.

Reply:
If I get rid of that conditional, then the default value for `wcombine` in MemDFJK will not be `true` because the global default is `false` . This default was chosen because the other JK classes don't have machinery to combine AO tensors.

Reply:
below is the goal, right? the read_options summary will need language that the default is most efficient available for the `scf_type` algorithm. earlier opposition to `if has_changed` above retracted. thanks for clearing away the driver-side options querying.

user `wcombine` | eff. `scf_type` | outcome
----|----|----
true | mem_df | omega combined
false | mem_df | omega separate
(unset) | mem_df | omega combined (b/c more efficient)
true | disk_df | NYI error
false | disk_df | omega separate
(unset) | disk_df | omega separate

Reply:
That is the expected, desired, and achieved behavior with this PR, yes.